src/header/mod.rs

```diff
@@ -3,16 +3,15 @@
 //! Information gleaned from [wikipedia](https://en.wikipedia.org/wiki/BMP_file_format) and
 //! [this website](http://paulbourke.net/dataformats/bmp/)
 
+use embedded_graphics::prelude::*;
```

Comment:
I have a feeling this was caused by a rust-analyzer autoimport, but either way I'd prefer not to use the prelude and/or wildcard imports.

Is it part of the standard order to have this, then `mod`, then `use crate`? That looks weird too but if it's part of the normal behaviour feel free to leave it.

Reply:
> I have a feeling this was caused by a rust-analyzer autoimport, but either way I'd prefer not to use the prelude and/or wildcard imports.

It wasn't caused by rust-analyzer. I think we had previously discussed this and decided to not use the prelude inside e-g itself but use the prelude in examples and external creates. But if you have a good reason for not using the prelude it won't take much to convince me to change it.

> Is it part of the standard order to have this, then `mod`, then `use crate`? That looks weird too but if it's part of the normal behaviour feel free to leave it.

It is a weird and not part of any standard. Maybe like this?
```rust
use embedded_graphics::prelude::*;

use crate::{
    color_table::ColorTable,
    parser::{le_u16, le_u32, take, take_slice},
    ParseError,
};

mod dib_header;

use dib_header::DibHeader;
```
This form makes it easier to add a `pub` to the imports from child modules if necessary. But I'm open to other suggestions.

Reply:
> decided to not use the prelude inside e-g itself but use the prelude in examples and external creates

Right, gotcha, let's leave it as is then.

> Maybe like this?

I like putting `mod` before any imports but that's just me. I'm happy to use your suggestion though. Either way let's not bikeshed this too much :)

```
mod dib_header;

use embedded_graphics::prelude::*;

use crate::{
    color_table::ColorTable,
    parser::{le_u16, le_u32, take, take_slice},
    ParseError,
};

use dib_header::DibHeader;
```

Reply:
> I like putting `mod` before any imports but that's just me. I'm happy to use your suggestion though. Either way let's not bikeshed this too much :)

It is bikeshedding, but I would prefer to use one common style across all e-g crates. I've used my last suggestion for now and adjusted all other imports in the crate to match.

My reason for putting the mods in between the imports is to divide imports from other crates from child module imports. Otherwise it's not as clear that `use dib_header::DibHeader` isn't importing something from a `dib_header` crate. If it is preceded by `mod dib_header` it should be obvious that it's an import from a child module.